src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -5264,6 +5388,105 @@ static bool TakeOutParam(object y, out int x)
             VerifyNotInScope(model, x7Ref[2]);
         }
 
+        [Fact]
+        public void Scope_FieldInitializers_04()
+        {
+            var source =
+@"
+class X : Y
+{
+    public static void Main()
+    {
+    }
+
+    bool Test3 = TakeOutParam(out int x3, x3);
+
+    bool Test4 = TakeOutParam(out var x4, x4);
+
+    bool Test5 = TakeOutParam(out int x5, 5);
+
+    X()
+    : this(x5)
+    {
+        System.Console.WriteLine(x5);
+    }
+
+    X(object x) 
+    : base(x5) 
+    => System.Console.WriteLine(x5);
+
+    static bool Test6 = TakeOutParam(out int x6, 6);
+    static X()
+    {
+        System.Console.WriteLine(x6);
+    }
+
+    static bool TakeOutParam(out int x, int y) 
+    {
+        x = 123;
+        return true;
+    }
+}
+
+class Y
+{
+    public Y(object y) {}
+}
+";
```

Comment:
Are we testing references (presumably invalid) across fields?
```
static bool Test7 = TakeOutParam(out x6, 7);
```

Reply:
> Are we testing references (presumably invalid) across fields?

See in Scope_FieldInitializers_01
```
    bool Test71 = TakeOutParam(7, out int x7) && x7 > 0; 
    bool Test72 = Dummy(x7, 2); 
```

---
In reply to: [175110611](https://github.com/dotnet/roslyn/pull/25519#discussion_r175110611) [](ancestors = 175110611)